test/builder/QuarkBuilderCometWithdraw.t.sol

```diff
@@ -373,4 +376,129 @@ contract QuarkBuilderCometWithdrawTest is Test, QuarkBuilderTest {
+
+        ChainPortfolio[] memory chainPortfolios = new ChainPortfolio[](1);
+        chainPortfolios[0] = ChainPortfolio({
+            chainId: 1,
+            account: address(0xa11ce),
+            nextNonce: 12,
+            assetSymbols: Arrays.stringArray("USDC", "USDT", "LINK", "WETH"),
+            assetBalances: Arrays.uintArray(0, 0, 0, 0),
+            cometPortfolios: cometPortfolios
+        });
+
+        QuarkBuilder builder = new QuarkBuilder();
+        QuarkBuilder.BuilderResult memory result = builder.cometWithdraw(
+            cometWithdraw_(1, cometUsdc_(1), "USDC", type(uint256).max),
+            chainAccountsFromChainPortfolios(chainPortfolios), // user has no assets
+            paymentUsdc_(maxCosts) // but will pay from withdrawn funds
+        );
+
+        address quoteCallAddress = CodeJarHelper.getCodeAddress(
+            abi.encodePacked(type(Quotecall).creationCode, abi.encode(ETH_USD_PRICE_FEED_1, USDC_1))
+        );
+        address cometWithdrawActionsAddress = CodeJarHelper.getCodeAddress(type(CometWithdrawActions).creationCode);
+
+        assertEq(result.paymentCurrency, "usdc", "usdc currency");
+
+        // Check the quark operations
+        assertEq(result.quarkOperations.length, 1, "one operation");
+        assertEq(
+            result.quarkOperations[0].scriptAddress,
+            quoteCallAddress,
+            "script address is correct given the code jar address on mainnet"
+        );
+        assertEq(
+            result.quarkOperations[0].scriptCalldata,
+            abi.encodeWithSelector(
+                Quotecall.run.selector,
+                cometWithdrawActionsAddress,
+                abi.encodeWithSelector(
+                    CometWithdrawActions.withdraw.selector, cometUsdc_(1), usdc_(1), type(uint256).max
+                ),
+                0.1e6
+            ),
+            "calldata is Quotecall.run(CometWithdrawActions.withdraw(COMET, USDC_1, uint256.max), 0.1e6);"
+        );
+        assertEq(
+            result.quarkOperations[0].expiry, BLOCK_TIMESTAMP + 7 days, "expiry is current blockTimestamp + 7 days"
+        );
+
+        // check the actions
+        assertEq(result.actions.length, 1, "one action");
+        assertEq(result.actions[0].chainId, 1, "operation is on chainId 1");
+        assertEq(result.actions[0].quarkAccount, address(0xa11ce), "0xa11ce sends the funds");
+        assertEq(result.actions[0].actionType, "WITHDRAW", "action type is 'WITHDRAW'");
+        assertEq(result.actions[0].paymentMethod, "QUOTE_CALL", "payment method is 'QUOTE_CALL'");
+        assertEq(result.actions[0].paymentToken, USDC_1, "payment token is USDC");
+        assertEq(result.actions[0].paymentMaxCost, 0.1e6, "payment max is set to 0.1e6");
+        assertEq(
+            result.actions[0].actionContext,
+            abi.encode(
+                Actions.WithdrawActionContext({
+                    amount: type(uint256).max, // ?? should this be a different amount?
```

Comment:
Good question. Let's check with :kelp: on this. Though there really isn't a way for QuarkBuilder to get the exact amount because of interest accrual.

Reply:
@mykelp can you confirm that it is acceptable to return the `WithdrawActionContext` with `uint256.max` as the value for a max withdraw?

Reply:
yeah, that makes sense. We can handle it appropriately in the view